packages/@internationalized/number/src/NumberParser.ts

```diff
@@ -215,7 +215,10 @@ function getSymbols(formatter: Intl.NumberFormat, intlOptions: Intl.ResolvedNumb
   let pluralLiterals = allParts.filter(p => !nonLiteralParts.has(p.type)).map(p => escapeRegex(p.value));
   let singularLiterals = singularParts.filter(p => !nonLiteralParts.has(p.type)).map(p => escapeRegex(p.value));
   let sortedLiterals = [...new Set([...singularLiterals, ...pluralLiterals])].sort((a, b) => b.length - a.length);
-  let literals = new RegExp(`${sortedLiterals.join('|')}|[\\p{White_Space}]`, 'gu');
+  let literals = new RegExp('[\\p{White_Space}]', 'gu');
+  if (sortedLiterals.length >= 1) {
+    literals = new RegExp(`${sortedLiterals.join('|')}|[\\p{White_Space}]`, 'gu');
```

Comment:
Seems like it's probably compiling an extra regex in the common case (where there are literals). Maybe turn this into a ternary?

Reply:
right, it had a leading `'|'` with nothing preceding it, which wouldn't match anything